### ISSUE TITLE
50 EUR Donation Autocompleting Relationship Fields With Default Value Yield Into Strange Behaviour

### ISSUE THREAD

**Issue by Mythli** [FrozenNode/Laravel-Administrator#835]
# What is happening?

Strange Javascript errors which render the laravel administrator unworking.

![screen shot 2015-03-09 at 19 41 24](https://cloud.githubusercontent.com/assets/504049/6562177/1fda84c4-c697-11e4-9487-67d8409f9191.png)

# How to reproduce the issue?

Reproducing the Issue is very easy. Just create a relationship field with autocomplete turned on and set a default value like this:

```
'creator' => [
			'type' => 'relationship',
			'title' => 'Ersteller',
			'name_field' => 'email',
			'value' => 1,
			'autocomplete' => true
		],
```

# How to fix this Issue?

After some debugging I figured out what may cause the issue. In the Select2RemoteHandler Binding a data attribute is build using the text representation of the currently selected value.

The default value is just not included in the json response. Therefore retrieving the corresponding key fails in the following line:

```
data = {id: val, text: admin.viewModel[options.field + '_autocomplete'][val].text};
```

I fixed this with some dubious code in the getModelRelatedItems function in Model\Config. This fixed the Issue for non editable fields. However, the Issue still persists for editable relationship fields.

```
if($items->count() < 1 && $autocomplete && $value) {
				 return [$model->{$name}()->getRelated()->find($value)];
}
```
I donate 50 euros via paypal to anyone who fixes this Issue. The only requirement is, that the fix stays open source and can be integrated into this repository.

Best Regards
Tobias Anhalt

**Comment by Mythli** [comment]
moved code segment to the top

**Comment by RemiCollin** [comment]
Ok, this one was tricky.. :) 

There are in fact two separate causes to this bug. 

1) Setting a default value to the the dropdown in autocomplete mode will have the initSelect method call the admin object which is not initialized yet. a quick check on the state of the admin object fix it.

2) If autocomplete is set on true, relationship options are not loaded in the initial data, (which is of course the desired behaviour as we want to load them on autocomplete). But that also mean our default value will not be available in the dropdown.
So, in order to use a default value, we can fetch the object corresponding to the value in config, and add it to the 'autocomplete' array, when initilializing the viewModel.

I added a PR with the bugfix : https://github.com/FrozenNode/Laravel-Administrator/pull/836

**Comment by janhartigan** [comment]
@RemiCollin I'll take a look at this sometime in the next couple of days.

**Comment by davemathews** [comment]
@RemiCollin  check that it is fixed now in v5.0.5 and open a new ticket if you still have issues.  thanks.
